File: archive/c/c-sharp/JosephusProblem.cs
if (
    args is not [var nText, var kText]
    || !int.TryParse(nText, out int n)
    || !int.TryParse(kText, out int k)
    || n <= 0
    || k <= 0
)
{
    Console.Error.WriteLine(
        "Usage: please input the total number of people and number of people to skip."
    );
    return;
}

int survivor = 0;
for (int i = 2; i <= n; i++)
    survivor = (survivor + k) % i;

Console.WriteLine(survivor + 1);

File: archive/v/visual-basic/josephus-problem.vb
Module JosephusProblem

    Public Sub Main(args As String())

        If args.Length < 2 Then
            Console.WriteLine("Usage: please input the total number of people and number of people to skip.")
            Return
        End If

        Dim n As Integer
        Dim k As Integer

        If Not Integer.TryParse(args(0), n) OrElse
           Not Integer.TryParse(args(1), k) OrElse
           n <= 0 OrElse k <= 0 Then

            Console.WriteLine("Usage: please input the total number of people and number of people to skip.")
            Return
        End If

        Dim survivor As Integer = Josephus(n, k)

        Console.WriteLine(survivor)

    End Sub


    Private Function Josephus(n As Integer, k As Integer) As Integer

        Dim result As Integer = 0

        For m As Integer = 2 To n
            result = (result + k) Mod m
        Next

        Return result + 1

    End Function

End Module